modules/apps/commerce/commerce-frontend-js/src/main/resources/META-INF/resources/utilities/forms/formsHelper.js

```diff
@@ -74,7 +75,11 @@ export function updateFields(currentFields, nextField) {
 		const {fieldName} = fieldInstance;
 
 		if (fieldName === key) {
-			nextFields.push({key: fieldName, value: toArray(nextValue)});
+			nextFields.push({
+				key: fieldName,
+				required: currentField['required'],
```

Comment:
Just a note for the next time (feel free to keep it this way) ;) 

```
required: currentField['required']
```
can be written as
```
required: currentField.required
```